docs/Getting Started/Arch Linux/Arch Linux Root on ZFS.rst

```diff
@@ -874,52 +870,46 @@ Optional Configuration
       -o keyformat=raw \
       rpool_$INST_UUID/sys
 
-  #. Remove ``zfsencryptssh`` hook.
-     Encrypted boot pool is incompatible with
-     password by SSH::
-
-      sed -i 's|zfsencryptssh||g' /etc/mkinitcpio.conf
-
-     If ``zfsencryptssh`` is not removed, initrd will
-     stuck at ``fail to load key material`` and fail to boot.
-
-  #. Generate initrd::
+  #. Import encrypted boot pool from ``/dev/mapper``::
 
-      mkinitcpio -P
-
-  #. As keys are stored in initrd,
-     set secure permissions for ``/boot``::
-
-      chmod 700 /boot
-
-  #. Import boot pool after starting systemd::
-
-       tee /etc/systemd/system/zfs-bpool_$INST_UUID-import-cache.service <<EOF
+       tee /etc/systemd/system/zfs-import-bpool-mapper.service <<EOF
        [Unit]
-       Description=Import boot pool by cache file
+       Description=Import encrypted boot pool
        Documentation=man:zpool(8)
        DefaultDependencies=no
        Requires=systemd-udev-settle.service
-       After=zfs-import-cache.service
-       After=zfs-import.target
+       After=cryptsetup.target
        Before=boot.mount
-       ConditionFileNotEmpty=/etc/zfs/zpool.cache
        ConditionPathIsDirectory=/sys/module/zfs
 
        [Service]
        Type=oneshot
        RemainAfterExit=yes
-       ExecStart=/usr/bin/zpool import -c /etc/zfs/zpool.cache -aN
```

Comment:
Personally, I don't love the cache file. But it does still seem to be the preferred approach, regardless of how often people have suggested deprecating it. If you're doing this **only** because you saw someone say it was deprecated, you should probably not. If you're doing this intentionally and knowingly, I won't stand in your way. However, make sure you have well tested having a third pool (i.e. something other than bpool or rpool).

Reply:
I deprecated it because zpool.cache causes major headaches when embedded in initrd.

FYI, Arch Linux uses mkinitcpio to generate initrd, and zfs support is added by [a hook](https://github.com/archzfs/archzfs/blob/master/src/zfs-utils/zfs-utils.initcpio.hook). Its behavior is, embed zpool.cache in initramfs if file exists, and import **only** from the embedded zpool.cache file at boot, skipping scanning entirely. If the vdev path or configuration has been changed, and initrd has not been regenerated, initrd will still import the pool in the old vdev configuration on next reboot, which you know, not good.

Example: recently I sold by laptop, [bricked while testing Secure Boot](https://github.com/openzfs/openzfs-docs/pull/127#issuecomment-786976799). I took the NVMe drive out and put it in an external USB3 enclosure to boot it from another computer. This enclosure contains a RealTek controller, which converts the NVMe disk into an ATA one. In `/dev/disk/by-id`, this changed the device name from `nvme-SAMSUNG-...` to `ata-nvme-SAMSUNG-...`. Of course, thanks to the stale cache, rpool failed to import and the system ends up in kernel panic. One must boot up live CD and chroot into the broken system to delete the stale cache, regenerate initrd and force import-by-scanning on next reboot.

Reply:
Sounds like you're doing so intentionally then, so that's fine. Did you test having a third (non-bpool, non-rpool) pool?

Reply:
Yes, just tested on my server. The third pool imports fine with import-by-scanning.

Reply:
The pool are imported with `-N`, thus not mounted by initrd. The mounting is handled by zfs-mount.service for third pool, zfs-mount-generator for rpool, fstab for bpool.

Reply:
Boot dataset is a tricky one because while it must be must be mounted at boot, it also must be `canmount=noauto` to support boot environments. zfs-mount-generator will not mount `canmount=noauto` at boot, so `mountpoint=legacy` and fstab are used instead.